### ISSUE TITLE
How to implement LSTM/GRU language model through Keras?

### ISSUE THREAD

**Issue by Sanqiang** [fchollet/keras#5360]
I want to implement LSTM/GRU language model by Keras. But it seems one challenge.
How to generate softmax over all the vocabularies? I.e. I cannot access weight in Embedding layer in either my own implemented loss function as well as function in Lambda layer.

**Comment by siddheshk** [comment]
I think I didn't understand this completely. You do know the number of words in your vocab beforehand right?
So why not define a dense layer like
```python
Dense(num_vocab)(some_layer)
```
This will generate softmax over all the words in the vocab. You can also define your loss function over this output. Please do correct me if I am wrong.

Cheers.

**Comment by Sanqiang** [comment]
Actually, I want to say is I want to generate Softmax over embedding of vocabulary.
Say, Input layer only can input the target word index target_i, and through the Embedding layer, I am able to get embedding of the target_i, embed(target_i).
But I want to take the Softmax over all the embedding of words, say embed(target_i) / sum (embed(all words)). 
In order to maximize the softmax, I need my own implemented loss function to access all the embedding of vocabulary (denominator of softmax). Is it feasible?

**Comment by amy12xx** [comment]
Could you learn the embeddings as a separate model (as a feature transformer) that becomes the input to the LSTM/GRU model, or do you need to train the embedding weights as well as part of the Language Model?

**Comment by Sanqiang** [comment]
@amy12xx 
I want to train the embedding through the LSTM/GRU. So the input of LSTM/GRU will be the embedding of a word and tune the embedding through the LSTM/GRU.
